src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtils.java

```diff
@@ -371,6 +410,147 @@ private static Properties readConfigFileProperties(final Path configFilePath) {
+                        }
+                    }
+
+                    if (alternateSource == null && line.startsWith(README_ALT_SOURCE_LINE_START)) {
+                        final Matcher m = ALT_SOURCE_PATTERN.matcher(line);
+                        if ( m.matches() ) {
+                            alternateSource = m.group(1);
+                        }
+                        else {
+                            logger.warn("README file has improperly formatted alternate source string: " + line);
+                        }
+                    }
+
+                    line = reader.readLine();
+                }
+
+                // Make sure we have good info:
+                if ( version == null ) {
+                    logger.warn("Unable to read version information from data sources info/readme file: " + readmePath.toUri().toString());
+                }
+                else {
+                    logger.info("Data sources version: " + version);
+
+                    // Make sure our version is OK:
+                    dataSourcesPathIsAcceptable = validateVersionInformation(versionMajor, versionMinor, versionYear, versionMonth, versionDay);
+                }
+
+                if ( source == null ) {
+                    logger.warn("Unable to read source information from data sources info/readme file: " + readmePath.toUri().toString());
+                }
+                else {
+                    logger.info("Data sources source: " + source);
+                }
+
+                if ( alternateSource == null ) {
+                    logger.warn("Unable to read alternate source information from data sources info/readme file: " + readmePath.toUri().toString());
+                }
+                else {
+                    logger.info("Data sources alternate source: " + alternateSource);
+                }
+            }
+            catch (final Exception ex) {
+                logger.warn("Could not read " + README_FILE_NAME + ": unable to log data sources version information.");
+            }
+        }
+        else {
+            logger.warn("Could not read " + README_FILE_NAME + ": unable to log data sources version information.");
+        }
+
+        // Warn the user if they need newer stuff.
+        if ( !dataSourcesPathIsAcceptable ) {
+            logger.error("ERROR: Given data source path is too old!  Minimum required version is: " + CURRENT_MINIMUM_DATA_SOURCE_VERSION + " (yours: " + version + ")");
+            logger.error("       You must download a newer version of the data sources from the Broad Institute FTP site: " + DATA_SOURCES_FTP_PATH);
+            logger.error("       or the Broad Institute Google Bucket: " + DATA_SOURCES_BUCKET_PATH);
+        }
+
+        return dataSourcesPathIsAcceptable;
+    }
+
+    @VisibleForTesting
+    static boolean validateVersionInformation(final int major, final int minor, final int year, final int month, final int day) {
```

Comment:
No need for `final` on primitives for method parameters.

Reply:
Sounds good.

Reply:
Except that the codebase does not do this elsewhere.